agent/api/task.go

```diff
@@ -432,9 +433,11 @@ func (task *Task) dockerHostConfig(container *Container, dockerContainerMap map[
 		}
 	}
 
-	err = task.platformHostConfigOverride(hostConfig)
-	if err != nil {
-		return nil, &HostConfigError{err.Error()}
+	if cfg.TaskCPUMemLimit {
```

Comment:
If we are going to simply override other host config fields too in the future (as the method name suggests), having this check inside platformHostConfigOverride method makes more sense.

Reply:
I was hesitant about passing `cfg` down to the platformOverrides but I think its a better call for future overrides.